LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/test_generate_emissions.py

```diff
@@ -74,39 +92,56 @@ def test_001_same_start_end_date_behavior(
     sim_sd, sim_ed, n_sims = mock_source_and_args_for_generate_emissions_same_st_ed[1]
     expected_res: date = mock_source_and_args_for_generate_emissions_same_st_ed[2]
     for sim in range(n_sims):
-        src.generate_emissions(sim_start_date=sim_sd, sim_end_date=sim_ed, sim_number=sim)
+        src.generate_emissions(
+            sim_start_date=sim_sd,
+            sim_end_date=sim_ed,
+            sim_number=sim,
+            emission_rate_source_dictionary={
+                "test": EmissionsSourceSample("test", "gram", "second", [1], 1000)
+            },
+            repair_delay_dataframe={},
+        )
     emis_counts: list[int] = []
     for sim, emissions in src._generated_emissions.items():
         emis_counts.append(len(emissions))
     assert average(emis_counts) == pytest.approx(expected_res, rel=0.1)
 
 
-# TODO : fix up below test
 # def test_001_validate_randomness(self):
```

Comment:
This was my bad. But we may want to fix up this test.

Reply:
Can take a look